tools/depends/target/ffmpeg/Makefile

```diff
@@ -42,6 +42,7 @@ $(PLATFORM): $(DEPS) | $(TARBALLS_LOCATION)/$(ARCHIVE).$(HASH_TYPE)
 	rm -rf $(PLATFORM); mkdir -p $(PLATFORM)
 	rm -rf $(PLATFORM); mkdir -p $(PLATFORM)/build
 	cd $(PLATFORM); $(ARCHIVE_TOOL) $(ARCHIVE_TOOL_FLAGS) $(TARBALLS_LOCATION)/$(ARCHIVE)
+	cd $(PLATFORM); patch -p1 -i ../0001-avformat-install-demux.h.patch
```

Comment:
I cant see how this is going to work for us. This puts a hard link on using "our" ffmpeg. This header wont be available from a linux system package, and we dont force usage of "our" internal lib version.

The private api usage has to be changed imo

Reply:
I agree.

How much do we care about these special wav files that actually contain DTS or SPDIF (ac3?) inside them? These files can also play fine without the special handling but in that case they won't passthrough but are decoded using ffmpeg instead. I needed to search for 15 minutes to even find a wav file with dts inside bringing up the question if we should just drop passing through these files instead. For the spdif inside wav I did not find any sample file on the internet. Also the advanced setting `VideoPlayerignoredtsinwav` is not documented anywhere on the wiki.

I think for the other case where we access `read_seek` we can do something similar as in https://github.com/FFmpeg/FFmpeg/commit/50f34172e0cca2cabc5836308ec66dbf93f5f2a3.

Reply:
FFmpeg's demuxer can detect the DTS nicely, which is infact what we are using here. Which function call is missing that demux.h exposes?

Input #0, wav, from 'dts_diatonis_secret-universe.wav':
  Duration: 00:06:47.56, bitrate: 1411 kb/s
  Stream #0:0: Audio: dts (DTS) ([1][0][0][0] / 0x0001), 44100 Hz, 5.0(side), fltp, 1411 kb/s

Reply:
We were accessing the read [`read_probe`](https://github.com/FFmpeg/FFmpeg/blob/master/libavformat/demux.h#L63) function of the `dts` / `spdif` decoder which is not accessible anymore. 

I don't think probing the formats is manually is not needed anymore. As @fritsch pointed out ffmpeg correctly detects a dts stream inside a wav container. If the code path in question is removed my AVR still switches to dts and passes through the dts audio. I still don't have a sample file for the spdif (ac3?) use case though.

Reply:
Could you check that kodi actually still opens a passthrough sink? As playing the WAV with PCM without internal volume might actually produce the same result :P but involves some danger.

We have some not-so-smart people who encode dts wavs to flac ... and yeah ... expect kodi to decode and bit transform losslessly ;-) so that the AVR still understands DTS ... while we only open a PCM sink.

Reply:
Yes seems fine:
```
2024-04-12 22:30:00.940735+0200 Kodi[85210:9873580] 2024-04-12 22:30:00.940 T:9873580   debug <general>: GetAEChannelMap Engine requests layout RAW, RAW
2024-04-12 22:30:00.940773+0200 Kodi[85210:9873580] 2024-04-12 22:30:00.940 T:9873580   debug <general>: GetAEChannelMap trying to map to stereo layout: [Left] [Right]
2024-04-12 22:30:00.940798+0200 Kodi[85210:9873580] 2024-04-12 22:30:00.940 T:9873580   debug <general>: GetAEChannelMap mapped channels to layout FL, FR
2024-04-12 22:30:00.940884+0200 Kodi[85210:9873580] 2024-04-12 22:30:00.940 T:9873580   debug <general>: Initialize: using buffer size: 24576 (0.139320 ms)
2024-04-12 22:30:00.953121+0200 Kodi[85210:9873580] 2024-04-12 22:30:00.953 T:9873580   debug <general>: CActiveAESink::OpenSink - DARWINOSX Initialized:
2024-04-12 22:30:00.953156+0200 Kodi[85210:9873580] 2024-04-12 22:30:00.953 T:9873580   debug <general>:   Output Device : DENON-AVR
2024-04-12 22:30:00.953182+0200 Kodi[85210:9873580] 2024-04-12 22:30:00.953 T:9873580   debug <general>:   Sample Rate   : 44100
2024-04-12 22:30:00.953200+0200 Kodi[85210:9873580] 2024-04-12 22:30:00.953 T:9873580   debug <general>:   Sample Format : AE_FMT_S16NE
2024-04-12 22:30:00.953217+0200 Kodi[85210:9873580] 2024-04-12 22:30:00.953 T:9873580   debug <general>:   Channel Count : 2
2024-04-12 22:30:00.953237+0200 Kodi[85210:9873580] 2024-04-12 22:30:00.953 T:9873580   debug <general>:   Channel Layout: FL, FR
2024-04-12 22:30:00.953255+0200 Kodi[85210:9873580] 2024-04-12 22:30:00.953 T:9873580   debug <general>:   Frames        : 1536
2024-04-12 22:30:00.953273+0200 Kodi[85210:9873580] 2024-04-12 22:30:00.953 T:9873580   debug <general>:   Frame Size    : 4
```

Reply:
Let's assume this code is historical from a time where ffmpeg could not find DTS in wav files - and ditch it ... we are early in the cycle and will find out during upcoming months.